SimpleExec/Command.cs

```diff
@@ -90,22 +90,23 @@ public static async Task RunAsync(string name, string args = null, string workin
         /// <param name="echoPrefix">The prefix to use when echoing the command line and working directory (if specified) to standard error (stderr).</param>
         /// <param name="configureEnvironment">An action which configures environment variables for the command.</param>
         /// <param name="createNoWindow">Whether to run the command in a new window.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> to observe while waiting for the command to exit.</param>
         /// <returns>A <see cref="string"/> representing the contents of standard output (stdout).</returns>
         /// <exception cref="NonZeroExitCodeException">The command exited with non-zero exit code.</exception>
         /// <remarks>
         /// By default, the resulting command line and the working directory (if specified) are echoed to standard error (stderr).
         /// To suppress this behavior, provide the <paramref name="noEcho"/> parameter with a value of <c>true</c>.
         /// </remarks>
-        public static string Read(string name, string args = null, string workingDirectory = null, bool noEcho = false, string windowsName = null, string windowsArgs = null, string echoPrefix = null, Action<IDictionary<string, string>> configureEnvironment = null, bool createNoWindow = false)
+        public static string Read(string name, string args = null, string workingDirectory = null, bool noEcho = false, string windowsName = null, string windowsArgs = null, string echoPrefix = null, Action<IDictionary<string, string>> configureEnvironment = null, bool createNoWindow = false, CancellationToken cancellationToken = default)
         {
             using (var process = new Process())
             {
                 process.StartInfo = ProcessStartInfo.Create(name, args, workingDirectory, true, windowsName, windowsArgs, configureEnvironment, createNoWindow);
 
-                var runProcess = process.RunAsync(noEcho, echoPrefix ?? DefaultPrefix.Value, CancellationToken.None);
+                var runProcess = process.RunAsync(noEcho, echoPrefix ?? DefaultPrefix.Value, cancellationToken);
                 var readOutput = process.StandardOutput.ReadToEndAsync();
 
-                Task.WaitAll(runProcess, readOutput);

```

Comment:
Upon cancellation, WaitAll throws an AggregateException containing an inner exception as OperationCanceledException.

```
> await Task.WhenAll(Task.FromCanceled(new CancellationToken(true)));

> System.Threading.Tasks.TaskCanceledException: A task was canceled.
Task.WaitAll(Task.FromCanceled(new CancellationToken(true)));
> System.AggregateException: One or more errors occurred.
  + System.Threading.Tasks.Task.WaitAll(System.Threading.Tasks.Task[], int, System.Threading.CancellationToken)
  + System.Threading.Tasks.Task.WaitAll(System.Threading.Tasks.Task[], int)
  + System.Threading.Tasks.Task.WaitAll(System.Threading.Tasks.Task[])
```

so when doing sync over async `GetAwaiter().GetResult()` is our evil friend :(